codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2ConnectionDecoder.java

```diff
@@ -233,35 +232,32 @@ void onUnknownFrame0(ChannelHandlerContext ctx, byte frameType, int streamId, Ht
         @Override
         public int onDataRead(final ChannelHandlerContext ctx, int streamId, ByteBuf data,
                 int padding, boolean endOfStream) throws Http2Exception {
-            // Check if we received a data frame for a stream which is half-closed
             Http2Stream stream = connection.requireStream(streamId);
```

Comment:
@nmittler - Great point.

Lets handle this in a separate issue https://github.com/netty/netty/issues/3557.